### ISSUE TITLE
is hyperparameter tuning needed for better performance?

### ISSUE THREAD

**Issue by christk1** [davidsandberg/facenet#1005]
hello, i trained my model with triplet loss on my own aligned dataset(not faces) with near 5000 ids for about 110 epochs.Training loss went from ≈4 down to ≈0.3 . The arguments are:

model_def: models.inception_resnet_v2
batch_size: 90
image_size: 160
ids_per_batch: 45
images_per_id: 40
epoch_size: 1000
alpha: 0.2
embedding_size: 128
random_crop: False
random_flip: False
keep_probability: 1.0
weight_decay: 0.0001
optimizer: RMSPROP
learning_rate: 0.01
learning_rate_decay_epochs: 100
learning_rate_decay_factor: 1.0
moving_average_decay: 0.9999
seed: 666
learning_rate_schedule_file: data/learning_rate_schedule.txt
vall_nrof_folds: 10

most classes had one image so i did data augmentation and the minimum images per class became 40.For validation is done in 300 from 5000 ids.

the results are the following:

step   train_accuracy  vall_rate
0  	        0.68833	0.04233
1169	0.73033	0.06633
2321	0.76217	0.06967
3464	0.74400	0.10933
4606	0.77417	0.06100
5750	0.78600	0.10100
6892	0.76833	0.09700
8030	0.79683	0.06700
9168	0.77667	0.08233
10305	0.80533	0.11533
11434	0.82067	0.16300
12562	0.81150	0.13733
13690	0.81517	0.17200
14819	0.78817	0.09133
15947	0.84217	0.13800
17069	0.81000	0.13033
18200	0.82933	0.13967
19314	0.83500	0.19500
20440	0.82917	0.17433
21564	0.84417	0.18700
22694	0.84650	0.18200
23809	0.83700	0.12967
24926	0.83117	0.16567
26044	0.86683	0.25733
27160	0.84717	0.19533
28280	0.85917	0.25767
29393	0.84517	0.15067
30504	0.85017	0.15567
31611	0.85367	0.26467
32715	0.84467	0.27267
33829	0.83383	0.26733
34943	0.86833	0.25300
36060	0.86183	0.21733
37164	0.85267	0.23067
38277	0.86183	0.22967
39366	0.85700	0.23700
40473	0.85417	0.13867
41581	0.86850	0.28067
42687	0.84633	0.23300
43786	0.86283	0.23667
44889	0.86817	0.31300
45990	0.85283	0.34633
47082	0.87683	0.28633
48187	0.87683	0.29067
49284	0.86267	0.26100
50391	0.86550	0.19700
51487	0.88783	0.24467
52601	0.86250	0.30533
53708	0.88417	0.30500
54822	0.87817	0.24167
55915	0.84233	0.22467
57008	0.87767	0.27900
58100	0.87117	0.28700
59200	0.87250	0.22633
60297	0.86017	0.27067
61399	0.87067	0.29767
62486	0.86150	0.28167
63593	0.86983	0.25500
64700	0.84217	0.15967
65803	0.87400	0.22767
66902	0.88533	0.24033
67996	0.87183	0.26700
69091	0.88917	0.27200
70185	0.86433	0.26933
71278	0.88450	0.37533
72371	0.88617	0.27133
73463	0.87800	0.28167
74558	0.88367	0.32900
75651	0.88233	0.31600
76741	0.88850	0.32800
77824	0.88650	0.29633
78896	0.88617	0.27500
79986	0.87300	0.28633
81072	0.86133	0.21533
82153	0.87933	0.29233
83248	0.87617	0.28633
84330	0.85117	0.31100
85428	0.88050	0.35033
86540	0.86817	0.33500
87646	0.88300	0.24500
88750	0.86400	0.19500
89849	0.87200	0.28700
90952	0.86683	0.22567
92042	0.88350	0.36200
93130	0.88467	0.33400
94191	0.87317	0.38000
95297	0.88733	0.37567
96378	0.86717	0.24500
97456	0.86417	0.27733
98537	0.88417	0.39133
99628	0.88400	0.39100
100712	0.88467	0.37400
101792	0.88083	0.39567
102889	0.87583	0.37700
103980	0.86500	0.37133
105066	0.89317	0.43100
106152	0.89133	0.45267
107222	0.88050	0.30367
108304	0.89367	0.37933
109394	0.87650	0.33567
110492	0.88167	0.27800
111578	0.88600	0.31800

the best result for vallidation rate is 0.45267. The loss (0.3) for 110 epochs is very low, my question is, should i tune my hyperparameters to see improvement or redo my data augmentation and retrain or both or change training method? I made this issue because i have limited resources for training and it is my first triplet loss training. What are your suggestions?

**Comment by anandhakrishnanh** [comment]
@christk1  Did you get any good solutions

**Comment by christk1** [comment]
Yes the problem was my messy dataset

**Comment by anandhakrishnanh** [comment]
Can you tell me how did you clean your dataset so that the problem was resolved.
